saleor/core/permissions/__init__.py

```diff
@@ -0,0 +1,111 @@
+# flake8: noqa
+
+from django.contrib.auth import get_user_model
+
+from .auth_filters import (
+    AuthorizationFilters,
+    is_app,
+    is_staff_user,
+    is_user,
+    resolve_authorization_filter_fn,
+)
+from .enums import (
+    PERMISSIONS_ENUMS,
+    AccountPermissions,
+    AppPermission,
+    BasePermissionEnum,
+    ChannelPermissions,
+    CheckoutPermissions,
+    DiscountPermissions,
+    GiftcardPermissions,
+    MenuPermissions,
+    OrderPermissions,
+    PagePermissions,
+    PageTypePermissions,
+    PaymentPermissions,
+    PluginsPermissions,
+    ProductPermissions,
+    ProductTypePermissions,
+    ShippingPermissions,
+    SitePermissions,
+    get_permission_names,
+    get_permissions,
+    get_permissions_codename,
+    get_permissions_enum_dict,
+    get_permissions_enum_list,
+    get_permissions_from_codenames,
+    get_permissions_from_names,
+    split_permission_codename,
+)
+
+
+def one_of_permissions_or_auth_filter_required(context, permissions):
+    """Determine whether user or app has rights to perform an action.
+
+    The `context` parameter is the Context instance associated with the request.
+    """
+    if not permissions:
+        return True
+
+    authorization_filters = [
+        p for p in permissions if isinstance(p, AuthorizationFilters)
+    ]
+    permissions = [p for p in permissions if not isinstance(p, AuthorizationFilters)]
+
+    granted_by_permissions = False
+    granted_by_authorization_filters = False
+
+    # TODO: move this function from graphql to core
```

Comment:
Should be solved in this PR?

Reply:
I'll do it separately as this PR is already large.